src/units.cpp

```diff
@@ -185,8 +185,10 @@ bool updateUnitMultiplier(const UnitsPtr &units, int direction, double &multipli
             units->unitAttributes(i, ref, pre, exp, expMult, id);
             mult = std::log10(expMult);
 
-            if (isStandardPrefixName(pre)) {
-                prefixMult = standardPrefixList.at(pre);
+            // Make sure convertPrefixToInt is safe to use whilst trying to avoid
+            // causing an exception.
+            if (pre.empty() || isStandardPrefixName(pre) || isCellMLInteger(pre)) {
+                prefixMult = convertPrefixToInt(pre);
             } else {
                 return false;
             }
```

Comment:
I was expecting something like (not tested):
```c++
bool ok;
prefixMult = convertPrefixToInt(pre, &ok);
if (!ok) {
    return false;
}
```
with
```c++
int convertPrefixToInt(const std::string &in, bool *ok)
{
    int prefixInt = 0;
    if (isStandardPrefixName(in)) {
        prefixInt = standardPrefixList.at(in);
    } else if (in.empty()) {
        prefixInt = 0;
    } else if (isCellMLInteger(in) {
        convertToInt(in, prefixInt);
    } else if (ok != nullptr) {
        ok = false;
    }

    return prefixInt;
}
```

Reply:
My preference is to keep `convertPrefixToInt` as it is.  What do you think @nickerso?

Reply:
You can still call `convertPrefixToInt()` the way you were doing it before, it's just that now you don't have to do those ugly tests before calling it in `updateUnitMultiplier()`. This makes code maintenance unnecessarily "complex".

Reply:
True, but I am also looking at the other locations that this function is used and that proposal doesn't work well for those cases.

Reply:
I've already approved this one twice, so I think its pretty clear I'm happy with it as is....

Reply:
What other locations? Re `convertPrefixToInt()`, if you want the exact same behaviour as your version, then you could have:
```c++
int convertPrefixToInt(const std::string &in, bool *ok)
{
    int prefixInt = 0;
    if (isStandardPrefixName(in)) {
        prefixInt = standardPrefixList.at(in);
    } else if (in.empty()) {
        prefixInt = 0;
    } else if ((ok == nullptr) || isCellMLInteger(in) {
        convertToInt(in, prefixInt);
    } else if (ok != nullptr) {
        ok = false;
    }

    return prefixInt;
}
```

Reply:
`convertPrefixToInt` is used in the analyser and validator right.

Reply:
Indeed, they use it and I have just tested my implementation at https://github.com/cellml/libcellml/pull/846#discussion_r618845525 and the one at https://github.com/cellml/libcellml/pull/846#discussion_r618896294, and it all works fine. So, yes, I would use my implementation at https://github.com/cellml/libcellml/pull/846#discussion_r618845525.

Reply:
FWIW `convertPrefixToInt()` can be optimised a bit (since we initialise `prefixInt` to `0`:
```c++
int convertPrefixToInt(const std::string &in, bool *ok)
{
    int prefixInt = 0;

    if (ok != nullptr) {
        *ok = true;
    }

    if (isStandardPrefixName(in)) {
        prefixInt = standardPrefixList.at(in);
    } else if (isCellMLInteger(in)) {
        convertToInt(in, prefixInt);
    } else if ((ok != nullptr) && !in.empty()) {
        *ok = false;
    }

    return prefixInt;
}
```

Reply:
7ebbfc9